ipam/spec.go

```diff
@@ -0,0 +1,21 @@
+package ipam
+
+import "context"
+
+type Interface interface {
+	// Test executes the cluster IPAM test using the configured provider
+	// implementation. The test processes the following steps to
+	// ensure the provider specific operator implements guest cluster IPAM
+	// correctly.
+	//
+	//  - Create guest clusters #1, #2, #3.
+	//	- Wait for guest clusters to be ready.
+	//	- Verify that clusters have distinct subnets.
+	//  - Terminate guest cluster #2 and create guest cluster #4.
```

Comment:
Immediately. This is the key aspect which we want to verify because this is the case which fails in production right now.

Reply:
Ack.